lldb/source/Commands/CommandObjectDWIMPrint.cpp

```diff
@@ -61,6 +63,36 @@ void CommandObjectDWIMPrint::HandleArgumentCompletion(
 
 bool CommandObjectDWIMPrint::DoExecute(StringRef command,
                                        CommandReturnObject &result) {
+  // BEGIN SWIFT
+  // Dump the contents of the valobj to result, possibly prepending it with
+  // a note telling the user that p is potentially better.
+  auto write_valobj_to_result =
+      [&](ValueObject &valobj, const DumpValueObjectOptions &dump_options) {
+        // Identify the default output of po "<Name: 0x...>.
+        // The regex is:
+        // - Start with "<".
+        // - Followed by 1 or more non-whitespace characters.
+        // - Followed by ": 0x".
+        // - Followed by 1 or more characters in the 0-9 or a-f range.
+        // - Followed by ">".
+        static const std::regex swift_class_regex("^<[^\\s]+: 0x[0-9a-f]+>");
```

Comment:
```suggestion
        static const std::regex swift_class_regex("<\\S+: 0x[[:xdigit:]]{5,}>\\s*");
```

* `[^\s]` -> `\S`
* `[0-9a-f]` to `[[:xdigit:]]`, or alternatively `[0-9a-fA-F]`, or use `icase` when constructing the regex
* use `{5,}` instead of `+` for the hex repetition (though this is up for debate, maybe it should still match `0x08` or `0x2040` for example)
* append `\s*` so that `std::regex_match` can be used instead of `regex_search`

Reply:
Should the regex end in `$` to make sure no additional contents is there?

Reply:
Just FYI, we also have https://llvm.org/doxygen/classllvm_1_1Regex.html